core-compose/src/main/java/com/squareup/workflow/ui/compose/WorkflowContainer.kt

```diff
@@ -0,0 +1,323 @@
+import androidx.ui.savedinstancestate.savedInstanceState
+import com.squareup.workflow.Snapshot
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.compose.ComposeRendering
+import com.squareup.workflow.diagnostic.WorkflowDiagnosticListener
+import com.squareup.workflow.launchWorkflowIn
+import com.squareup.workflow.ui.ViewEnvironment
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.CONFLATED
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import okio.ByteString
+import kotlin.coroutines.CoroutineContext
+
+// TODO this is probably WAY too many overloads
+
+/**
+ * Render a [Workflow]'s renderings.
+ *
+ * When this function is first composed it will start a new runtime. This runtime will be restarted
+ * any time [workflow], [diagnosticListener], or the `CoroutineContext`
+ * changes. The runtime will be cancelled when this function stops composing.
+ *
+ * @param workflow The [Workflow] to render.
+ * @param props The props to render the root workflow with. If this value changes between calls,
+ * the workflow runtime will re-render with the new props.
+ * @param onOutput A function that will be invoked any time the root workflow emits an output.
+ * @param diagnosticListener A [WorkflowDiagnosticListener] to configure on the runtime.
+ * @param content A [Composable] function that gets executed every time the root workflow spits
+ * out a new rendering.
+ */
+@Composable fun <PropsT, OutputT : Any, RenderingT> WorkflowContainer(
+  workflow: Workflow<PropsT, OutputT, RenderingT>,
+  props: PropsT,
+  onOutput: (OutputT) -> Unit,
+  modifier: Modifier = Modifier,
+  diagnosticListener: WorkflowDiagnosticListener? = null,
+  content: @Composable() (rendering: RenderingT) -> Unit
+) {
+  @Suppress("DEPRECATION")
+  val rendering =
+    observeWorkflow(workflow, props, onOutput, CoroutineContextAmbient.current, diagnosticListener)
+
+  Box(modifier = modifier) {
+    content(rendering)
+  }
+}
+
+/**
+ * Render a [Workflow]'s renderings.
+ *
+ * When this function is first composed it will start a new runtime. This runtime will be restarted
+ * any time [workflow], [diagnosticListener], or the `CoroutineContext`
+ * changes. The runtime will be cancelled when this function stops composing.
+ *
+ * @param workflow The [Workflow] to render.
+ * @param onOutput A function that will be invoked any time the root workflow emits an output.
+ * @param diagnosticListener A [WorkflowDiagnosticListener] to configure on the runtime.
+ * @param content A [Composable] function that gets executed every time the root workflow spits
+ * out a new rendering.
+ */
+@Direct
+@Composable fun <OutputT : Any, RenderingT> WorkflowContainer(
```

Comment:
Not sure if the compose compiler supports this yet, but try making these overloads inline.